packages/interbit/src/chainManagement/destroyRemovedChains.js

```diff
@@ -0,0 +1,45 @@
+const {
+  coreCovenant: {
+    actionCreators: { destroy }
+  },
+  manifest: {
+    selectors: { getRootChainId, getManifest, getChainAliases, getChains }
+  }
+} = require('interbit-covenant-tools')
+
+/**
+ * Destroys the chains that have been removed from the manifest file.
+ * This is done by comparing the manifest in the root chain's state
+ * to the manifest passed into params. Any that are in root chain state
+ * but are not in params are destroyed if this cli has access to them.
+ * This function should be called prior to setting a new manifest on the
+ * root chain.
+ * @param {Object} cli - The cli to destroy chains on
+ * @param {Object} manifest - The new manifest to compare with the root manifest
+ */
+const destroyRemovedChains = (cli, manifest) => {
+  const rootChainId = getRootChainId(manifest)
+  const rootChainInterface = cli.getChain(rootChainId)
+  const rootChainState = rootChainInterface.getState()
+  const rootChainManifest = getManifest(rootChainState)
+
+  if (!rootChainManifest) {
+    return
+  }
+
+  const chainAliasesConfiguredOnRoot = getChainAliases(rootChainManifest)
+  const chainEntries = Object.entries(getChains(manifest))
+  const removedChains = chainEntries.filter(
+    chainEntry => !chainAliasesConfiguredOnRoot.includes(chainEntry[0])
+  )
+
+  for (const removedChain of removedChains) {
+    const removedChainInterface = cli.getChain(removedChain)
```

Comment:
Something is not quite right here. The values in `removedChains` returned from filtering `Object.entries`  are key/value pairs (you filtering them using the chain alias from `chainEntry[0]`). Then I would expect the corresponding expression for getting the chain to be `cli.getChain(removedChain[1])` as that would be the chain ID. Alternatively deconstruct the chain alias and chain ID in the right spots.

Please be careful when naming variables that contain key/value pairs (aka `entries`) so we don't fall into these traps. I am also not keen on the naming `removedChain` and `removedChains` as I really don't know what type of object is being referred to (elsewhere in the code base, this is the chain interface).

Reply:
Good eye, this should be caught in integrations tests when all the open issues are finalised in the cascading-deploy. To be doubly sure, I have opened a ticket for a wide ranging integration test that incorporates destroy #658